pkg/executor/client/job.go

```diff
@@ -542,9 +542,6 @@ func (c *JobExecutor) TailJobLogs(ctx context.Context, id string, logs chan []by
 				if err = wait.PollImmediate(pollInterval, c.podStartTimeout, executor.IsPodLoggable(ctx, c.ClientSet, pod.Name, c.Namespace)); err != nil {
 					l.Errorw("poll immediate error when tailing logs", "error", err)
 					return err
-				} else if err = wait.PollImmediate(pollInterval, pollTimeout, executor.IsPodReady(ctx, c.ClientSet, pod.Name, c.Namespace)); err != nil {
-					l.Errorw("poll immediate error when tailing logs", "error", err)
```

Comment:
Why are we removing it?

Reply:
the logic is tricky PollImediate will return true when the func IsPodLoggable returns true but the error is nil so it will go to the next PollImmediate, IsPodReady is waiting for corev1.PodSucceeded and it will not let the streaming start.

Reply:
1. We have this logic in 2 places, if we want to change it
2. Second check was added to support cases of wrong images or not reachable at the moment, because in this case we simply doesn't have any errors and logs. So need to find a way to work properly for both things